pyresample/utils/_cf.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 Pyresample developers
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+def _load_crs_from_cf(nc_handle, grid_mapping_varname):
+    """ use pyproj to parse the content of the grid_mapping variable and initialize a crs object """
+    from pyproj import CRS
+    # here we assume the grid_mapping_varname exists (checked by caller)
+    return CRS.from_cf(vars(nc_handle[grid_mapping_varname]))
+
+
+def _is_valid_coordinate_variable(nc_handle, coord_varname, axis):
+    """ check if a coord_varname is a valid CF coordinate variable """
+    if axis not in ('x', 'y'):
+        raise ValueError("axis= parameter must be 'x' or 'y'")
+    coord_var = nc_handle[coord_varname]
+    valid = False
+    try:
+        valid = getattr(coord_var, 'standard_name') == 'projection_'+axis+'_coordinate'
+    except AttributeError:
+        valid = False
+    return valid
+
+
+def _load_axis_extent(nc_handle, coord_varname):
+    """ load extent for the axis held in coord_varname (min, max, len) """
+
+    # this requires reading the data
+    values = nc_handle[coord_varname][:]
+    try:
+        unit = values.units
+    except AttributeError:
+        unit = 'default'
+
+    # spacing
+    #   TBC: a) should we check the axis has constant spacing or does CF impose this already ?
+    #         b) is a single diff enough, or should we take a mean of several (rounding error) ?
+    spacing = abs(values[1] - values[0])
+
+    # extent (0,5*spacing is because area_def expects corner coordinates of the corner cells,
+    #     while CF stores center coords)
+    extent_low = spacing.min() - 0.5 * spacing
+    extent_hgh = spacing.max() + 0.5 * spacing
+
+    # now we take into account the units
+    scalef = 1.
+    if unit != 'default':
+        if unit == 'km':
+            scalef = 1000.
+        elif unit == 'm':
+            scalef = 1.
+        else:
+            raise ValueError("Sorry: un-supported unit!")
+
+    extent_low *= scalef
+    extent_hgh *= scalef
+
+    return extent_low, extent_hgh, len(values)
+
+
+def load_cf_area(nc_file, variable=None, y=None, x=None, ):
```

Comment:
This function is very long, and with comments you have already split it in 3 parts. Could these parts be a function each ?

Reply:
maybe :) I now re-shuffled most of the code, so I close your comment for now.